backend/src/index.js

```diff
@@ -40,114 +41,123 @@ BODY
-
-        const redirRes = await axios.get(`https://www.signingsavvy.com/${redirect}`);
-        const redirBody = await redirRes.data;
-
-        const $$ = cheerio.load(redirBody);
-        const link = $$(`#main_content_left > div.content_module > div > div.signing_body > div > link`).attr().href;
-
-        return link;
-    }
-    else
-    {
-        const link = $(`#main_content_left > div.content_module > div > div.signing_body > div > link`).attr().href;
-        return link;
+    const access = $(
+      `#main_content_left > div.content_module > div.search_results`
+    );
+    if (access.length) {
+      const redirect = access
+        .children()
+        .first()
+        .children()
+        .first()
+        .children()
+        .first()
+        .attr().href;
+
+      const redirRes = await axios.get(
+        `https://www.signingsavvy.com/${redirect}`
+      );
+      const redirBody = redirRes.data;
+
+      const $$ = cheerio.load(redirBody);
+      const link = $$(
+        `#main_content_left > div.content_module > div > div.signing_body > div > link`
+      ).attr().href;
+
+      return link;
+    } else {
+      const link = $(
+        `#main_content_left > div.content_module > div > div.signing_body > div > link`
+      ).attr().href;
+      return link;
     }
+  } catch (err) {
+    return undefined;
+  }
 }
 
-async function downloadVideoToServer(videoUrl, filePath) 
-{
-    try 
-    {
-        const finished = promisify(stream.finished);
-        const writer = createWriteStream(filePath);
-        return axios({
-            method: 'get',
-            url: videoUrl,
-            responseType: 'stream',
-        }).then(response => {
-            response.data.pipe(writer);
-            return finished(writer);
-        });
-    } 
-    catch (err)
-    {
-        console.log(err);
-    }
+async function downloadVideoToServer(videoUrl, filePath) {
+  try {
+    const finished = promisify(stream.finished);
+    const writer = createWriteStream(filePath);
+    return axios({
+      method: "get",
+      url: videoUrl,
+      responseType: "stream",
+    }).then((response) => {
+      response.data.pipe(writer);
+      return finished(writer);
+    });
+  } catch (err) {
+    console.log(err);
+  }
 }
 
-async function downloadVideo(url, word)
-{
-    try 
-    {
-        const videoUrl = await getVideoLink(url);
-        const filePath = path.join("./src", "..", "videos", `${word}.mp4`);
-        await downloadVideoToServer(videoUrl, filePath);
-        return `./videos/${word}.mp4`;
-    } 
-    catch (err) 
-    {
-        console.log(err);
-        return null;
-    }
+async function downloadVideo(url, word) {
+  try {
+    const videoUrl = await getVideoLink(url);
+    const filePath = path.join("./src", "..", "videos", `${word}.mp4`);
+    await downloadVideoToServer(videoUrl, filePath);
+    return `./videos/${word}.mp4`;
+  } catch (err) {
+    console.log(err);
+    return null;
+  }
```

Comment:
Ngl, this is inefficient af. Throw it in a worker thread for funzies. Paths can be dealt with some sort of resource manager or utility functions.